src/ir/particle_spec_test.cc

```diff
@@ -37,39 +37,34 @@ struct ParticleSpecProtoAndExpectedInfo {
 class ParticleSpecFromProtoTest :
    public testing::TestWithParam<ParticleSpecProtoAndExpectedInfo> {
  protected:
-  ParticleSpecFromProtoTest()
-    : particle_spec_registry_(),
-      particle_spec_(CreateParticleSpec(particle_spec_registry_)) {}
+  ParticleSpecFromProtoTest() : particle_spec_(CreateParticleSpec()) {}
 
-  static const ParticleSpec &CreateParticleSpec(
-      ParticleSpecRegistry &particle_spec_registry) {
+  static std::unique_ptr<ParticleSpec> CreateParticleSpec() {
     arcs::ParticleSpecProto particle_spec_proto;
```

Comment:
It looks like you've undone the pattern that I had where constructing a `ParticleSpec` required passing in the context object that would hold it (formerly `ParticleSpecRegistry`, now `SystemSpec`) so that its `unique_ptr` could not be held by any object other than the context object. I wrote this pattern to preserve the guarantee that it would definitely be held by an object with a long enough lifetime, but I do recognize that it was a little awkward to work with in tests. Is that why you've refactored it to make the `Create` method just return the owning pointer?

Reply:
OK, sounds good, that sounds fine to me.